### PR TITLE
Tanay unittests

### PR DESCRIPTION
**Pull request by tkulkarni97**
1. Refactored the 'train.py' script to make it unit-testable, for example:- added functions, extra arguments to it
2. Added the unit-testing script to it, the coverage gained is as below:
Screen Shot 2022-12-12 at 12 58 58 PM
3. I haven't added unit-testing cover and refactoring to the inference block in this PR, as first I want to know the uploading process of a blueprint to dev and production cluster.

### CODE REVIEW

**Review by jedwards-habana — APPROVED** [review]
LGTM

File: train/unittest_train.py [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+        ''' Checks the labels generated '''
+        self.assertListEqual(
+            sorted(["POSITIVE", "NEGATIVE", "NEUTRAL"]), sorted(self.labels)
+        )
+        
+    def test_model_predictions(self):
+        ''' Checks the model performance metrics - loss and accuracy '''
+        # Checks the model accuracy
+        self.assertTrue(
+            0 < self.score[1] <= 1
```
Comment (orionsBeltWest) [inline train/unittest_train.py]: Need to set the range tighter.  Say between 0.7 and 1.0 

0 for lower bound is too loose so everything is guaranteed to pass.

File: train/unittest_train.py [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+        )
+        # Checks the model loss
+        self.assertTrue(
+            0 < self.score[0] 
+        )
+
+    def test_model_history(self):
+        ''' Checks if loss and accuracy are returned for all of the epochs specified above '''
+        self.assertEqual(
+            len(self.acc), self.epochs_val
```
Comment (orionsBeltWest) [inline train/unittest_train.py]: let us discuss these I am not clear why comparing all the epochs_val ?

File: train/unittest_train.py [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+            len(self.loss), self.epochs_val
+        )
+        self.assertEqual(
+            len(self.val_loss), self.epochs_val
+        )
+
+    def test_model_history_metrics(self):
+        ''' Checks the average performance metrics from history '''
+        self.assertTrue(
+            0 < sum(self.acc) / len(self.acc) <= 1
```
Comment (orionsBeltWest) [inline train/unittest_train.py]: same here the lower bound is (0) which too loose. 

Also is this an average? or something else?

**Review by orionsBeltWest — CHANGES_REQUESTED** [review]
Looks good but  I need clarification about 2 test cases 

Let's discuss

File: train/unittest_train.py (hunk 3 above)
Comment (tkulkarni97) [reply train/unittest_train.py]: @orionsBeltWest Yes, I guess last value is good (already done)

File: train/unittest_train.py (hunk 2 above)
Comment (tkulkarni97) [reply train/unittest_train.py]: Check if we have same number of accuracy values for epochs

File: train/unittest_train.py (hunk 1 above)
Comment (tkulkarni97) [reply train/unittest_train.py]: @orionsBeltWest Added tighter bounds for this

**Review by orionsBeltWest — APPROVED** [review]
looks good.

**Pull request merged**